runtime/gc_base/StringTable.cpp

```diff
@@ -623,7 +630,12 @@ j9gc_createJavaLangString(J9VMThread *vmThread, U_8 *data, UDATA length, UDATA s
 
 		if (isASCII) {
 			for (UDATA i = 0; i < length; ++i) {
-				hash = (hash << 5) - hash + data[i];
+				U_8 c = data[i];
+				if (0xC0 == c) {
```

Comment:
A comment would be helpful (somewhere) that explains why we don't need to check the second byte is 0x80 here and several places below.